### PR TITLE
Refactoring code regarding SonarCloud

### LINKED ISSUE #231: Refactoring code regarding SonarCloud
#### Description
Refactoring code regarding SonarCloud

#### Tasks
- [ ] Check SonarCloud report and resolve priority high remark

### PR DESCRIPTION
**Pull request by pro100user19**
#### Description
Refactoring code regarding SonarCloud

#### Tasks
- Check SonarCloud report and resolve priority high remark

### CODE REVIEW

**Review by dshma — APPROVED** [review]

File: src/main/java/com/softserve/controller/ArchiveController.java [hunk 1]
```diff
@@ -44,13 +44,13 @@ public ArchiveController(ArchiveService archiveService, ScheduleService schedule
     @PostMapping("/{semesterId}")
     @ApiOperation(value = "Save archive schedule by semesterId in mongo db")
     @PreAuthorize("hasRole('MANAGER')")
-    public ResponseEntity archiveScheduleBySemester(@PathVariable("semesterId") Long semesterId) {
+    public ResponseEntity<Object> archiveScheduleBySemester(@PathVariable("semesterId") Long semesterId) {
```
Comment (dshma) [inline src/main/java/com/softserve/controller/ArchiveController.java:47]: why not the actual return type? ScheduleFullForArchiveDTO

File: src/main/java/com/softserve/controller/AuthenticationController.java [hunk 2]
```diff
@@ -53,14 +53,14 @@ public AuthenticationController(AuthenticationManager authenticationManager, Jwt
 
     @PostMapping("/sign-in")
     @ApiOperation(value = "Get credentials  for login")
-    public ResponseEntity signIn(@RequestBody AuthenticationRequestDTO requestDto) {
+    public ResponseEntity<Object> signIn(@RequestBody AuthenticationRequestDTO requestDto) {
```
Comment (dshma) [inline src/main/java/com/softserve/controller/AuthenticationController.java:56]: why not the actual return type? AuthenticationResponseDTO

File: src/main/java/com/softserve/controller/UserController.java [hunk 3]
```diff
@@ -108,7 +108,7 @@ public ResponseEntity<List<UserDTO>> getAllUsersWithRoleUser() {
 
     @GetMapping("/profile")
     @ApiOperation(value = "Get current user data")
-    public ResponseEntity getCurrentUser(@CurrentUser JwtUser jwtUser) {
+    public ResponseEntity<Object> getCurrentUser(@CurrentUser JwtUser jwtUser) {
```
Comment (dshma) [inline src/main/java/com/softserve/controller/UserController.java]: better to change to the actual return type

File: src/main/java/com/softserve/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -109,10 +109,11 @@ public User save(User object) {
     public User update(User object) {
         log.info("Enter into update method with entity:{}", object);
         getById(object.getId());
-        if (userRepository.findByEmail(object.getEmail()).isPresent() &&
-                !Objects.equals(userRepository.findByEmail(object.getEmail()).get().getId(), object.getId())) {
-            throw new FieldAlreadyExistsException(User.class, "email", object.getEmail());
-        }
+        userRepository.findByEmail(object.getEmail()).ifPresent(user -> {
+            if(!Objects.equals(user.getId(), object.getId())) {
```
Comment (dshma) [inline src/main/java/com/softserve/service/impl/UserServiceImpl.java]: do not forget to format the code you're adding

**Review by dshma — CHANGES_REQUESTED** [review]
do take account of the comments and checkstyle remarks

File: src/main/java/com/softserve/controller/ArchiveController.java (hunk 1 above)
Comment (pro100user19) [reply src/main/java/com/softserve/controller/ArchiveController.java:47]: In line 52 return ResponseEntity<MessageDTO>, and 64 line return ResponseEntity<TemporaryScheduleForArchiveDTO>. And so I return ResponseEntity<Object>.

File: src/main/java/com/softserve/controller/AuthenticationController.java (hunk 2 above)
Comment (pro100user19) [reply src/main/java/com/softserve/controller/AuthenticationController.java:56]: This is the situation with the previous comment.

File: src/main/java/com/softserve/repository/impl/RoomRepositoryImpl.java [hunk 5]
```diff
@@ -17,6 +17,11 @@
 @Slf4j
 public class RoomRepositoryImpl extends BasicRepositoryImpl<Room, Long> implements RoomRepository {
 
+    public static final String SEMESTER_ID = "semesterId";
+    public static final String DAY_OF_WEEK = "dayOfWeek";
+    public static final String CLASS_ID = "classId";
+    public static final String EVENT_ODD = "evenOdd";
```
Comment (dshma) [inline src/main/java/com/softserve/repository/impl/RoomRepositoryImpl.java]: typo: EVEN not EVENT

File: src/main/java/com/softserve/repository/impl/TemporaryScheduleRepositoryImpl.java [hunk 6]
```diff
@@ -42,9 +48,9 @@ public Long isExistTemporaryScheduleByVacationByDateWithIgnoreId(Long id, LocalD
                         "where  t.date = :date and t.vacation = :vacation and t.semester.id = :semesterId and t.id!= :id and t.period = null " +
                         "and t.teacher = null  and t.scheduleId = null ")
                 .setParameter("date", date)
```
Comment (dshma) [inline src/main/java/com/softserve/repository/impl/TemporaryScheduleRepositoryImpl.java]: why not creating a constant for this literal?

File: src/main/java/com/softserve/service/impl/StudentServiceImpl.java [hunk 7]
```diff
@@ -164,7 +166,7 @@ public Student delete(Student object) {
      * First line of the file is a header.
      * All subsequent lines contain data about students.
      * <p>
-     * "surname","name","patronymic","email"
+     * "surname","name","patronymic",EMAIL
```
Comment (dshma) [inline src/main/java/com/softserve/service/impl/StudentServiceImpl.java:168]: better to use literals in java doc rather than class variables

**Review by dshma — CHANGES_REQUESTED** [review]
It might be worth creating a Constants class as most of them are just class variable duplicates

File: src/main/java/com/softserve/repository/impl/TemporaryScheduleRepositoryImpl.java (hunk 6 above)
Comment (pro100user19) [reply src/main/java/com/softserve/repository/impl/TemporaryScheduleRepositoryImpl.java]: Okay I create

File: src/main/java/com/softserve/service/impl/StudentServiceImpl.java (hunk 7 above)
Comment (pro100user19) [reply src/main/java/com/softserve/service/impl/StudentServiceImpl.java:168]: I also think it will be better

**Review by RuslanSlobodian — CHANGES_REQUESTED** [review]
This applies to the previous closed PR, but since you have a fix "Technical borg previous PR" here, so
delete the string "!/Schedule_logs/log.log"
from .gitignore, because the whole line
"/schedule_logs!/schedule_logs/log.log" does not work.

File: src/main/java/com/softserve/util/ScheduleConstants.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
+package com.softserve.util;
+
+public final class ScheduleConstants {
```
Comment (dshma) [inline src/main/java/com/softserve/util/ScheduleConstants.java]: I don't think these are all schedule-related constants, thus "Schedule" appears a bit misleading, just Constants would be better

**Review by dshma — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserve/util/ScheduleConstants.java (hunk 8 above)
Comment (pro100user19) [reply src/main/java/com/softserve/util/ScheduleConstants.java]: Okay, I wanted to associate the class name with the project name

**Review by RuslanSlobodian — APPROVED** [review]

File: src/main/java/com/softserve/service/impl/TemporaryScheduleServiceImpl.java [hunk 9]
```diff
@@ -195,23 +192,23 @@ public void deleteTemporarySchedulesBySemesterId(Long semesterId) {
...
 
         EvenOdd keyOne = (countDays / 7) % 2 != 0 ? EvenOdd.EVEN : EvenOdd.ODD;
         EvenOdd keyTwo = keyOne.equals(EvenOdd.EVEN) ? EvenOdd.ODD : EvenOdd.EVEN;
         try {
             List<TemporarySchedule> one = getAllBySemesterAndRange(semesterId, from, from.plusDays(7));
             List<TemporarySchedule> two = getAllBySemesterAndRange(semesterId, from.plusDays(8), from.plusDays(14));
 
-            Map<DayOfWeek, List<TemporarySchedule>> oneMap = new HashMap<>();
-            Map<DayOfWeek, List<TemporarySchedule>> twoMap = new HashMap<>();
+            EnumMap<DayOfWeek, List<TemporarySchedule>> oneMap = new EnumMap<>(DayOfWeek.class);
```
Comment (dshma) [inline src/main/java/com/softserve/service/impl/TemporaryScheduleServiceImpl.java]: No need to change the reference type, it's a kinda rule of thumb to operate with abstraction on the left

File: src/main/java/com/softserve/service/impl/TemporaryScheduleServiceImpl.java [hunk 10]
```diff
@@ -195,23 +192,23 @@ public void deleteTemporarySchedulesBySemesterId(Long semesterId) {
...
+            EnumMap<DayOfWeek, List<TemporarySchedule>> twoMap = new EnumMap<>(DayOfWeek.class);
```
Comment (dshma) [inline src/main/java/com/softserve/service/impl/TemporaryScheduleServiceImpl.java]: same here

**Review by dshma — CHANGES_REQUESTED** [review]

**Review by dshma — APPROVED** [review]

**Pull request merged**